### PR TITLE
add documentation for make_jobs variable

### PR DESCRIPTION
**Pull request by mandresm**
Adds documentation for the `make_jobs` variable

### CODE REVIEW

File: lib/spack/docs/packaging_guide.rst [hunk 1]
```diff
@@ -2344,6 +2344,37 @@ you set ``parallel`` to ``False`` at the package level, then each call
...
+   :linenos:
+ 
+   class Xios(Package):
+      ...
+
+      def install(self, spec, prefix):
+         ...
+
+         options = [
+            "--full",
```
Comment (wdconinc) [inline lib/spack/docs/packaging_guide.rst]: I would only list the relevant arguments for this example. Using a real package is nice as it gives users a place to go for further details.

Comment (mandresm) [reply lib/spack/docs/packaging_guide.rst]: I removed all the arguments except the one passing `str(make_jobs)`

**Comment by mwkrentel** [comment]
In the array for `options`, I think you need to show `--job` for how
`str(make_jobs)` is used.  Not all the options, but those two go
together.  You could put them on one line in the interest of space:
```
    options = [
       ...
       "--jobs", str(make_jobs),
    ]
```

Also, I'd eliminate the blank lines after the three `...` lines,
again in the interest of space.

And maybe remove the final `...` and blank line after the `make_xios`,
it's not really needed.

Otherwise, looks good, thanks for writing this!

**Comment by mandresm** [comment]
@mwkrentel, you are welcome, I already addressed your suggestions :)

**Review by mwkrentel — APPROVED** [review]
I think you may need to adjust the `emphasize-lines` because some lines were deleted.
Otherwise, I approve.

**Comment by mandresm** [comment]
ups... I just fixed the line numbers now, sorry

**Review by wdconinc — APPROVED** [review]
Thanks a lot for this documentation update!

**Pull request merged**